WooCommerce/Classes/Authentication/Navigation Exceptions/ApplicationPasswordDisabledViewModel.swift

```diff
@@ -5,12 +5,20 @@ import WordPressAuthenticator
 /// modeling an error when application password is disabled.
 ///
 struct ApplicationPasswordDisabledViewModel: ULErrorViewModel {
-    init(siteURL: String) {
+    init(siteURL: String,
+         previousViewController: UIViewController?,
+         authentication: Authentication = ServiceLocator.authenticationManager) {
         self.siteURL = siteURL
+        self.previousViewController = previousViewController
+        self.authentication = authentication
     }
 
     let siteURL: String
-    let image: UIImage = .errorImage // TODO: update this if needed
+    let authentication: Authentication
+    let image: UIImage = .errorImage
+
+    // The VC that was showing before the application password flow. This is used to navigate back without guesswork.
+    let previousViewController: UIViewController?
```

Comment:
Related to a question above: the `previousViewController` reference is kept strongly here. Is it intentional to keep the weak reference in `ApplicationPasswordAuthorizationWebViewController` but strong here?

Reply:
As shared in the thread above, I think the previous view controller should not be deallocated while it's still in the navigation stack. It can only be released once you pop it from the stack.